cirq-core/cirq/ops/cxswap_czswap.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2024 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, Tuple
+
+import numpy as np
+
+import cirq
+from cirq.ops import gate_features, raw_types
+
+
```

Comment:
instead of overloading the operator yourself you can use
```suggestion
@value.value_equality()
```

this way you can remove __eq__ from the classes

Reply:
Thank you, I didn't know this existed!